[FE] 드롭다운메뉴 기능 & 좋아요 및 좋아요 취소

Jeonghae10 opened the pull request:
### 📕 Issue Number

Close #369 


### 📙 작업 내역

> 구현 내용 및 작업 했던 내역

- [x] 드롭다운메뉴 클릭 동작 
    - ex) 눌렀을 때 꺼지기, 메뉴 동작 후 꺼지기 등등.. 
- [x] 좋아요 및 좋아요 취소 동작
- [x] 드롭다운 - 내 플레이리스트에 추가
- [x] 드롭다운 - 가사보기
- [x] 체크박스 CSS 
- [x] 드롭다운 메뉴 커스텀 훅으로 생성


### 📘 작업 유형

- [x] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트


### 📋 체크리스트

- [x] Merge 하는 브랜치가 올바른가?
- [x] 코딩컨벤션을 준수하는가?
- [x] PR과 관련없는 변경사항이 없는가?
- [ ] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?
      <br/>

### 📝 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- 체크박스
  체크 박스 관련 commit은 css랑 기능 구현 시 삽질했던 내용이라 보지 않으셔도 될 것 같습니다...(merge branch 이전부분)
  이 부분 때문에 민지님 코드랑 충돌 날 것 같은데 두 분이 확인해주시면 CSS 부분 빼고 제가 직접 충돌 해결하고 머지하겠습니다.

- 드롭다운 메뉴 :  현재 플레이리스트 
  이 부분은 리덕스에 저장된 부분이 있어야 구현할 수 있을 것 같아 일단 구현하지 않았습니다

- api 데이터 형식이 수정되었습니다
   - 유저 보관함에 저장(좋아요)
    ```
      {
            id : trackId || albumId || ...
      }
     ``` 
   - 특정 플레이리스트에 앨범, 트랙 등 저장
    ```
    {
            id: playlistId,
            data: [ trackId || albumId || ....]
    }
   ```
<br/><br/>

Jeonghae10 committed:
[FE] fix: background-image 변경 및 이미지 위치 조정

- background image가 바뀜에 따라 이미지 위치 조정

Jeonghae10 committed:
[FE] feat: trackrow card 체크박스 선택시 trackrow card 배경색 변경 #346

- 트랙로우카드의 체크 박스 선택 시 트랙로우카드의 배경색 변경 동작 구

Jeonghae10 committed:
[FE] fix: 체크 박스 컴포넌트 porps 번경 #346

- 체크 박스 선택 여부에 따라 전체 체크박스 상태를 관리하기 위해 props 추가
- props 변경에 따른 변경사항 수정
  - storybook 코드 수정
  - 트랙로우카드에서 onChange 함수 porps로 전달

Jeonghae10 committed:
[FE] feat: 전체 체크박스 상태관리 및 props 로 추가 전달 #346

- 체크박스들을 관리하는 전체 체크박스의 상태를 관리하기 위해 useState 추가
- 상태에 따라서 다르게 props 전달

Jeonghae10 committed:
[FE] fix: checkbox props 추가 #346

- 전체 체크박스와 개별체크박스를 구별하기 위해 allCheck props 추가

Jeonghae10 committed:
Merge branch 'dev' of https://github.com/boostcamp-2020/Project01-B-User-Event-Collector into feat/etcFeature-fe

Jeonghae10 committed:
[FE] fix: 트랙로우 카드 setLike 및 드롭다운 메뉴 에러 수정 #369

- 트랙로우 카드 setLiked  타입에 맞게 수정
- 드롭다운 메뉴 MenuItem 파라미터 잘못 넘겨준 것 수정

Jeonghae10 committed:
[FE] fix: 상위컴포넌트에서 dropdown click handler 전달받도록 수정 #369

- 드롭다운 메뉴의 기능 동작 후 상태변화를 위해 상위 컴포넌트에서 클릭핸들러를 전달받을 수 있도록 props 및 interface type 수정

Jeonghae10 committed:
[FE] feat: dropdown menu 좋아요 및 가사보기 동작 구현 #369

- dropdown menu의 좋아요 동작 구현
 - 좋아요 클릭 시 서버에 좋아요 api 요청 : likeAction리
- dropdwon menu의 가사보기 동작 구현
- dropwon menu에 따라 다르게 동작하기 위해 handleClose에 case에 따라 분

Jeonghae10 committed:
[FE] feat: 좋아요 취소 동작 구현 #269

- 드롭다운 메뉴에서 좋아요 후 취소 동작 구현

Jeonghae10 committed:
[FE] feat: 좋아요/취소 api 요청 함수 및 url 추가 #369

- 좋아요/좋아요 취소 동작을 구현하기 위해 utils/api에 좋아요/좋아요취소 api 요청 함수 추가
- api 요청 보낼 주소 추가

Jeonghae10 committed:
[FE] feat: 드롭다운메뉴 - 내 플레이리스트에 추가 시 모달 창 보이기 #369

- 드롭다운메뉴에서 내 플레이리스트에 추가 클릭 시 플레이리스트를 보여주는 보달창이 보이도록 구현

Jeonghae10 committed:
[FE] feat: 드롭다운 메뉴 - 내 플레이리스트에 추가 기능동작 구현 #369

- 드롭다운메뉴에서 내 플레이리스트에 추가 클릭 시 해당 플레이리스트에 선택한 트랙을 저장하는 동작 구현

Jeonghae10 committed:
[FE] fix: 플레이리스트 모달 창의 플레이리스트카드에 id 추가 #369

- 내 플레이리스트에 저장하기 동작 구현시 해당 플레이리스트의 id 를 알기 위해 id 를 저장하는 component 추가

Jeonghae10 committed:
[FE] feat: 특정 플레이리스트에 트랙 추가 api 요청 함수 구현 및 url 저장 #369

- 특정 플레이리스트에 트랙을 저장하기 위해 api를 요청하는 함수 구현
- api 의 요청을 위한  url저장

Jeonghae10 committed:
[FE] refactor: apis 중복 코드 제거

- requestOptions에 대해 함수로 얻을 수 있도록 하여 중복된 코드 제거함

Jeonghae10 committed:
Merge branch 'dev' of https://github.com/boostcamp-2020/Project01-B-User-Event-Collector into feat/etcFeature-fe

Jeonghae10 committed:
[FE] feat: 드롭다운 메뉴 동작에 대한 hooks 생성 #369

- 다양한 드롭다운 메뉴에 대해서 재사용성을 위해 메뉴 동작에 대한 hooks을 생성

Jeonghae10 committed:
[FE] feat: 드롭다운 메뉴 종류에 대해 상수로 생성 #369

- 드롭다운 메뉴에 따라 동작을 달리 하기 위해 메뉴이름으로 분기
- 추후 확장성과 재사용성을 위해 메뉴 종류를 상수로 관리

Jeonghae10 committed:
[FE] fix: 드롭다운 메뉴 컴포넌트에 드롭다운액션 훅 적용 및 pops 수정 #369

- 새로 만든 드롭다운 메뉴 동작 훅을 드롭다운메뉴 컴포넌트에 적용
- 메뉴 동작에 따라 state를 변경해주기 위해 상위컴포넌트로 부터 변경할 state를 props 로 받음

Jeonghae10 committed:
[FE] fix: 드롭다운메뉴동작 훅 적용 후 좋아요 및 좋아요 취소 동작 확인 #369

- 드롭다운메뉴동작 훅 적용 후 좋아요 및 좋아요 취소 동작 확인
- 기존에 기능구현을 위해 작성됐던 코드 삭제

Jeonghae10 committed:
[FE] feat: 드롭다운메뉴액션 훅에서 가사보기, 내 플레이리스트에 추가 동작 구현 #369

- 드롭다운메뉴 액션으로 가사보기, 내플레이리스트에 추가 동작 구현
- 두 동작모두 모달 창을 띄움
- 내 플레이리스트에 추가는 유저 보관함의 플레이리스트 리스트를 가져옴

Jeonghae10 committed:
[FE] style: likeRequest, unlikeRequest 함수명 변경

- 보관함에 추가/삭제 동작이므로 likeRequest -> addToLibrary, unlikeRequest -> deleteFromRequest로 함수명 변경

Jeonghae10 committed:
[FE] feat: 드롭다운 메뉴 보관함에 추가 동작 구현 #369

- 드롭다운메뉴의 보관함에 추가 동작 구현

YimJiYoung commented on client/hooks/useDropDownAction.ts:
```diff
@@ -0,0 +1,59 @@
+    };
+    const handleClose = (e) => {
+        const content = e.currentTarget.innerText;
+        console.log('....?');
+        console.log(content);
```
console.log는 지워주시면 좋을 것 같아요  ~

YimJiYoung commented on client/utils/apis.js:
```diff
@@ -1,21 +1,47 @@
+    }
+};
+
+export const likeRequest = async (url, option) => {
```
postRequest가 더 적절하지 않을까요 ?? 현재 post api가 좋아요밖에는 없긴 하지만 ,,

Jeonghae10 replied:
post 가 또 있어서 일단 기능 위주로 해놨어요

YimJiYoung reviewed (approved):
드랍다운 훅으로 만드신거 좋네요 ~ 수고하셨습니다 !

mingd1023 commented on client/constants/dropDownMenu.ts:
```diff
@@ -0,0 +1,13 @@
+const dropDownMenu = {
+    like:'좋아요',
+    unlike:'좋아요 취소',
+    addToPlaylist:'내 플레이리스트에 추가',
```
드롭다운 메뉴 요소들 이렇게 항상 정리하면 좋겠다고 생각했는데 드디어 됐네요.....!!

mingd1023 commented on client/stores/CheckBox.tsx:
```diff
@@ -0,0 +1,15 @@
+import React, { useState } from 'react';
+
+export const CheckBoxContext = React.createContext(undefined);
```
이 파일 제거해주세요!!

mingd1023 commented on client/components/organisms/FloatingSelectMenu/index.tsx:
```diff
@@ -75,33 +75,47 @@ const ButttonAreaContainer = styled.div`
+    const [checkState, setCheckState] = useState({
+        checkedCnt: 0,
+        isAllChecked: false,
+    });
```
이 부분도 삭제하시는 걸까요??

Jeonghae10 replied:
넹 !! style 빼고 다 삭제 예정이에요!

mingd1023 reviewed (approved):
수고하셨습니다!

Jeonghae10 committed:
[FE] fix: 플레이리스트 모달창 드롭다운버튼 컴포넌트 안으로 이동 #369

- 내 플레이리스트 추가 시에 뜨는 플레이리스트 모달창을 드롭다운버튼으로 이동
- 기존 트랙로우카드 컴포넌트에 있던 플레이리스트 모달컴포넌트 관련 코드 삭제

Jeonghae10 committed:
[FE] feat: 추가할 플레이리스트 선택시 동작 구현 #369

- 내 플레이리스트에 추가 동작에서 추가할 플레이리스트 선택시 api 요청 동작 구현
- api 요청 완료 후 플레이리스트 모달 off

Jeonghae10 committed:
[FE] fix: 드롭다운 메뉴 prop 추가에 따른 수정 #369

- 드롭다운메뉴에 상태관리할 state props를 추가함에 따라 드롭다운메뉴를 사용하는 모든 컴포넌트에서 state를 넘겨줄 수 있도록 수정

Jeonghae10 committed:
[FE] style: 오타 및 띄어씌기 수정 #369

- 드롭다운 메뉴 컨텐츠 띄어쓰기 및 오타 수정

Jeonghae10 committed:
[FE] fix: 유저보관함에 저장 api 및 내 플레이리스트에 저장 api 명세서 변경에 따른 수정

- 저장 되어지는 데이터와 url만 달라지므로 함수로 만들어 재사용하기 위해 json key name 통일

Jeonghae10 committed:
Merge branch 'dev' into feat/etcFeature-fe

Jeonghae10 committed:
[FE] style: 필요없는 모듈 삭제

- 체크 박스 상태관리를 리덕스로 하면서 기존에 생성했던 체크박스 스토어 삭제

Jeonghae10 committed:
Merge branch 'feat/etcFeature-fe' of https://github.com/boostcamp-2020/Project01-B-User-Event-Collector into feat/etcFeature-fe
